test/framework/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreTestUtil.java

```diff
@@ -118,6 +123,27 @@ private static void assertIndexGenerations(BlobContainer repoRoot, long latestGe
         assertTrue(indexGenerations.length <= 2);
     }
 
+    private static void assertShardIndexGenerations(BlobContainer repoRoot, ShardGenerations shardGenerations) throws IOException {
+        final BlobContainer indicesContainer = repoRoot.children().get("indices");
+        for (IndexId index : shardGenerations.indices()) {
+            final List<String> gens = shardGenerations.getGens(index);
+            if (gens.isEmpty() == false) {
+                final BlobContainer indexContainer = indicesContainer.children().get(index.getId());
+                final Map<String, BlobContainer> shardContainers = indexContainer.children();
```

Comment:
should we also check which files should not be there?

Reply:
We never actually write any blobs to the root of the `indices` folder that's why there's no check for that. But sure I can add one (in a separate PR maybe since it's totally unrelated to this effort?) to make sure it stays that way :)

Reply:
Ah ok. See the TODO `assertConsistency`, I was gonna add some more inverse checks eventually for sure. This is mainly dependent on adding shard cleanups to the cleanup endpoint (we run it before the consistency check to make sure the indices folders match up in tests that do failure scenarios that leak files but don't corrupt the repo).
If we want to do these checks on the shard level we need to beef up the cleanup endpoint first or we'd have to exclude part of the check for failure scenarios. That's the only reason thats's not in here yet :)

Reply:
ok, makes sense